dist/static/site.js

```diff
@@ -0,0 +1,266 @@
+/** The name of the user */
+let name = null;
+
+/** The WebSocket connection */
+let sock = null;
+
+/** Heartbeat variable */
+let isAlive = true;
+
+function removeChildren(parent) {
+  while (parent.firstChild) {
+    parent.removeChild(parent.firstChild);
+  }
+}
+
+function createSong(song) {
+  const div = document.createElement('div');
+  div.className = 'song';
+
+  const divImage = document.createElement('div');
+  divImage.className = 'song-thumb';
+
+  const aThumb = document.createElement('a');
+  aThumb.href = song.webpage_url;
+  aThumb.target = '_blank';
+
+  const imgThumb = document.createElement('img');
+  imgThumb.src = song.thumbnail;
+
+  divImage.appendChild(imgThumb);
+
+  aThumb.appendChild(divImage);
+  div.appendChild(aThumb);
+
+  const songTitleDiv = document.createElement('div');
+  songTitleDiv.className = 'songtitle';
+  songTitleDiv.appendChild(document.createTextNode(song.title));
+
+  div.appendChild(songTitleDiv);
+
+  return div;
+}
+
+function createQueue(username, queues) {
+  const queueDiv = document.createElement('div');
+  queueDiv.className = 'queue';
+
+  const nameDiv = document.createElement('div');
+  const nameSpan = document.createElement('span');
+  nameSpan.className = 'name';
+  nameSpan.appendChild(document.createTextNode(username));
+  nameDiv.appendChild(nameSpan);
+  queueDiv.appendChild(nameDiv);
+
+  const songListDiv = document.createElement('div');
+  songListDiv.className = 'songlist';
+
+  const songs = queues[username];
+  for (let i = 0; i < songs.length; i++) {
+    const songDiv = createSong(songs[i]);
+
+    // remove button
+    if (username === name) {
+      const remoteBtn = document.createElement('button');
+      remoteBtn.className = 'b-remove';
+      (function (queuePos) { remoteBtn.onclick = () => { sock.send(`remove ${queuePos}`); }; }(i));
+      songDiv.appendChild(remoteBtn);
+    }
+    songListDiv.appendChild(songDiv);
+  }
+  queueDiv.appendChild(songListDiv);
+  return queueDiv;
+}
+
+function freeSongEntry() {
+  const inputSongURL = document.getElementById('input-songurl');
+  inputSongURL.disabled = false;
+  inputSongURL.value = '';
+
+  const songURLButton = document.getElementById('b-songurl');
+  songURLButton.disabled = false;
+}
+
+function checkIsAlive() {
+  if (!isAlive) {
+    document.getElementById('lostconn').style.display = 'block';
+  } else {
+    document.getElementById('lostconn').style.display = 'none';
+  }
+}
+
+function sendName(username) {
+  name = username; // Save username locally
```

Comment:
Validate name before sending a WebSocket request:
```javascript
if (username.length > 0 && username.length <= 20 && !username.match(/^[a-zA-Z\d]*/g)) return;
```

Reply:
Do you think its worth checking on the frontend as well? Guess it wouldn't hurt

Reply:
Agree with frontend validation, the backend can behave in unexpected ways and it’s a better design pattern to let the client validate input before sending it off